tests/test_integration.py

```diff
@@ -479,3 +484,82 @@ def circuit():
             return qml.expval(qml.PauliZ(wires=0))
 
         assert circuit() == -1
+
+class TestBatchExecution:
+    """Test the devices work correctly with the batch execution pipeline."""
+
+    @pytest.mark.parametrize("d", pldevices)
+    @pytest.mark.parametrize("shots", [None, 8192])
+    def test_one_qubit_circuit_batch_params(self, shots, d, backend, tol, mocker):
+        """Test that devices provide correct result for a simple circuit using
+        the batch_params transform."""
+        if (d[0] == "qiskit.aer" and "aer" not in backend) \
+          or (d[0] == "qiskit.basicaer" and "aer" in backend):
+            pytest.skip("Only the AerSimulator is supported on AerDevice")
+
+        if backend not in state_backends and shots is None:
+            pytest.skip("Hardware simulators do not support analytic mode")
+
+        dev = qml.device(d[0], wires=1, backend=backend, shots=shots)
+
+        # Batch the input parameters
+        batch_dim = 3
+        a = np.linspace(0, 0.543, batch_dim)
+        b = np.linspace(0, 0.123, batch_dim)
+        c = np.linspace(0, 0.987, batch_dim)
+
+        spy1 = mocker.spy(QiskitDevice, "batch_execute")
+        spy2 = mocker.spy(dev.backend, "run")
+
+        @qml.batch_params
+        @qml.qnode(dev)
+        def circuit(x, y, z):
+            """Reference QNode"""
+            qml.PauliX(0)
+            qml.Hadamard(wires=0)
+            qml.Rot(x, y, z, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(circuit(a, b, c), np.cos(a) * np.sin(b), **tol)
+
+        # Check that QiskitDevice.batch_execute was called
+        assert spy1.call_count == 1
+        assert spy2.call_count == 1
+
+    @pytest.mark.parametrize("d", pldevices)
+    @pytest.mark.parametrize("shots", [None, 8192])
+    def test_batch_execute_parameter_shift(self, shots, d, backend, tol, mocker):
+        """Test that devices provide correct result computing the gradient of a
+        circuit using the parameter-shift rule and the batch execution pipeline."""
+        if (d[0] == "qiskit.aer" and "aer" not in backend) \
+          or (d[0] == "qiskit.basicaer" and "aer" in backend):
+            pytest.skip("Only the AerSimulator is supported on AerDevice")
+
+        if backend not in state_backends and shots is None:
+            pytest.skip("Hardware simulators do not support analytic mode")
+
+        dev = qml.device(d[0], wires=3, backend=backend, shots=shots)
+
+        spy1 = mocker.spy(QiskitDevice, "batch_execute")
```

Comment:
Does it use `batch_execute` because of the parameter shift diff method?

Reply:
Yep! The parameter-shift computation goes through the batch execution pipeline. The exact number of executions will actually change slightly with the QNode change.